core/modules/search/search.module

```diff
@@ -105,19 +105,108 @@ function search_views_api() {
-    'info' => t('Search form'),
+    'info' => t('Search box'),
     'description' => t('The search form for searching site content.'),
   );
   return $blocks;
 }
 
+/**
+ * Implements hook_block_configure().
+ */
+function search_block_configure($delta = '', $settings = array()) {
+  $form = array();
+  if ($delta == 'form') {
+    $settings += array(
+      'label_display' => FALSE,
+      'label' => '',
+      'place_display' => FALSE,
+      'placeholder' => '',
+      'button_display' => TRUE,
+      'button' => 'Search',
+    );
+    $form['label_display'] = array(
+      '#type' => 'checkbox',
+      '#title' => t('Display a label before the search box.'),
+      '#default_value' => $settings['label_display'],
+    );
+    $form['label'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Label text'),
+      '#default_value' => $settings['label'],
+      '#states' => array(
+        'visible' => array(
+          ':input[name="block_settings[label_display]"]' => array('checked' => TRUE),
+        ),
+      ),
+    );
+    $form['place_display'] = array(
+      '#type' => 'checkbox',
+      '#title' => t('Add placeholder text inside the search box.'),
+      '#default_value' => $settings['place_display'],
+    );
+    $form['placeholder'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Placeholder text'),
+      '#default_value' => t($settings['placeholder']),
+      '#states' => array(
+        'visible' => array(
+          ':input[name="block_settings[place_display]"]' => array('checked' => TRUE),
+        ),
+      ),
+    );
+    $form['button_display'] = array(
+      '#type' => 'checkbox',
+      '#title' => t('Display the search button.'),
+      '#default_value' => $settings['button_display'],
+    );
+    $form['button'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Button text'),
+      '#default_value' => t($settings['button']),
+      '#states' => array(
+        'visible' => array(
+          ':input[name="block_settings[button_display]"]' => array('checked' => TRUE),
+        ),
+      ),
+    );
+  }
+  return $form;
+}
+
 /**
  * Implements hook_block_view().
  */
-function search_block_view($delta = '') {
+function search_block_view($delta = '', $settings = array(), $contexts = array()) {
   if (user_access('search content')) {
-    $block['subject'] = t('Search');
-    $block['content'] = backdrop_get_form('search_block_form');
+    $form = backdrop_get_form('search_block_form');
+
+    // Show the label.
+    if ($settings['label_display']) {
+      $form['search_block_form']['#title_display'] = 'before';
+
+      // Update the label text.
+      if ($settings['label'] != '') {
+        $form['search_block_form']['#title'] = t($settings['label']);
+      }
+    }
+
+    // Add a placeholder.
+    if ($settings['place_display'] && $settings['placeholder'] != '') {
+      $form['search_block_form']['#attributes']['placeholder'] = t($settings['placeholder']);
```

Comment:
and leave the `t()`?

Reply:
Yup, please leave the `t()` to allow translation of placeholder. But no need to `check_plain` because attributes are passed to [`backdrop_attributes`](https://api.backdropcms.org/api/backdrop/core%21includes%21common.inc/function/backdrop_attributes/1) who already call  `check_plain` .

Reply:
ok.